ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java

```diff
@@ -837,6 +844,118 @@ public static void checkColumnName(String columnName) throws SemanticException {
     return colList;
   }
 
+  /**
+   * Get the partition specs from the tree. This stores the full specification
+   * with the comparator operator into the output list.
+   *
+   * @return Map of partitions by prefix length. Most of the time prefix length will
+   *         be the same for all partition specs, so we can just OR the expressions.
+   */
+  public static Map<Integer, List<ExprNodeGenericFuncDesc>> getFullPartitionSpecs(
```

Comment:
can we find a new home for these 2 `static` methods? :)
`ql/src/java/org/apache/hadoop/hive/ql/parse/ParseUtils.java`

Reply:
Fixed!